libs/portals/admin/ids-admin/src/components/Applications/Applications.tsx

```diff
@@ -1,91 +1,233 @@
+import * as styles from './Applications.css'
+import { AuthApplicationsList } from './Applications.loader'
 
 const Applications = () => {
+  const originalApplications = useLoaderData() as AuthApplicationsList
   const { tenant } = useParams()
   const { formatMessage } = useLocale()
   const { setNavTitle } = useTenant()
 
+  const [applications, setApplications] = useState<AuthApplicationsList>(
+    originalApplications,
+  )
+  const [inputSearchValue, setInputSearchValue] = useState<string>('')
+
   useEffect(() => {
-    // TODO: Get application by id from backend
     setNavTitle(tenant ? tenant : formatMessage(m.tenants))
   })
 
-  return (
-    <GridContainer>
-      <GridRow rowGap={3}>
-        <Box display={'flex'} justifyContent={'spaceBetween'} columnGap={4}>
-          <Box>
+  const handleSearch = (value: string) => {
+    setInputSearchValue(value)
+
+    if (value.length > 0) {
+      const filteredList = applications.filter((application) => {
+        return (
+          application?.defaultEnvironment.displayName[0].value
+            .toLowerCase()
+            .includes(value.toLowerCase()) ||
+          application.applicationId.toLowerCase().includes(value.toLowerCase())
+        )
+      })
+      setApplications(filteredList)
+    } else {
+      setApplications(originalApplications)
+    }
+  }
+
+  const getHeader = ({ withCreateButton = true }) => {
+    return (
+      <GridRow rowGap={3} marginBottom={'containerGutter'}>
+        <Box
+          width={'full'}
+          display={'flex'}
+          justifyContent={'spaceBetween'}
+          columnGap={'gutter'}
+          alignItems={'center'}
+        >
+          <Stack space={2}>
             <Text variant={'h2'}>{formatMessage(m.applications)}</Text>
-          </Box>
-          <Box>
+            <Text variant={'default'}>
+              {formatMessage(m.applicationsDescription)}
+            </Text>
+          </Stack>
+          {withCreateButton && (
+            <Box>
+              <Button size={'small'}>Create Application</Button>
+            </Box>
+          )}
+        </Box>
+      </GridRow>
+    )
+  }
+
+  return applications.length === 0 ? (
+    <GridContainer>
+      {getHeader({ withCreateButton: false })}
+      <GridRow>
+        <Box
+          display={'flex'}
+          flexDirection={'column'}
+          border={'standard'}
+          borderRadius={'large'}
+          justifyContent={'center'}
+          alignItems={'center'}
+          padding={10}
+        >
+          <Text variant={'h3'}>No application created</Text>
+          <Text paddingTop={'gutter'}>
+            Lorem ipsum dolor sit amet consectetur. A non ut nulla vitae mauris
+            accumsan at tellus facilisi.
+          </Text>
+          <Box marginTop={6}>
             <Button size={'small'}>Create Application</Button>
           </Box>
+          <Box marginTop={'gutter'}>
+            <Button variant={'text'}>Learn more</Button>
```

Comment:
We should have all values prepared for translation. So it would be good to translate every string in `messages.ts`